synapseclient/models/agent.py

```diff
@@ -0,0 +1,447 @@
+            cloud_alias_id=self.cloud_alias_id,
+            synapse_client=syn,
+        )
+        return self.fill_from_dict(agent_response)
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Agent: {self.registration_id}"
+    )
+    async def get_async(self, *, synapse_client: Optional[Synapse] = None) -> "Agent":
+        """Gets an existing agent.
+
+        Arguments:
+            synapse_client: The Synapse client to use for the request. If None, the default client will be used.
+
+        Returns:
+            The existing Agent object.
+        """
+        syn = Synapse.get_client(synapse_client=synapse_client)
+        agent_response = await get_agent(
+            registration_id=self.registration_id,
+            synapse_client=syn,
+        )
+        return self.fill_from_dict(agent_response)
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Start_Agent_Session: {self.registration_id}"
+    )
+    async def start_session_async(
+        self,
+        *,
+        access_level: Optional[
+            AgentSessionAccessLevel
+        ] = AgentSessionAccessLevel.PUBLICLY_ACCESSIBLE,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "AgentSession":
+        """Starts an agent session.
+
+        Arguments:
+            access_level: The access level of the agent session.
+                Must be one of PUBLICLY_ACCESSIBLE, READ_YOUR_PRIVATE_DATA, or WRITE_YOUR_PRIVATE_DATA.
+                Defaults to PUBLICLY_ACCESSIBLE.
+            synapse_client: The Synapse client to use for the request.
+                If None, the default client will be used.
+
+        Returns:
+            The new AgentSession object.
+        """
+        access_level = AgentSessionAccessLevel(access_level)
+        syn = Synapse.get_client(synapse_client=synapse_client)
+        session = await AgentSession(
+            agent_registration_id=self.registration_id, access_level=access_level
+        ).start_async(synapse_client=syn)
+        self.sessions[session.id] = session
+        self.current_session = session.id
+        return session
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Agent_Session: {self.registration_id}"
+    )
+    async def get_session_async(
+        self, *, session_id: str, synapse_client: Optional[Synapse] = None
+    ) -> "AgentSession":
+        syn = Synapse.get_client(synapse_client=synapse_client)
+        session = await AgentSession(id=session_id).get_async(synapse_client=syn)
+        if session.id not in self.sessions:
+            self.sessions[session.id] = session
+        self.current_session = session.id
+        return session
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Prompt_Agent_Session: {self.registration_id}"
+    )
+    async def prompt(
+        self,
+        *,
+        session_id: Optional[str] = None,
+        prompt: str,
```

Comment:
Required args cannot show up after optional args. It should be fine to put this as the first argument